langgraph/pregel/__init__.py

```diff
@@ -1763,15 +1770,16 @@ def _proc_input(
     proc: PregelNode,
     managed: ManagedValueMapping,
     channels: Mapping[str, BaseChannel],
-    catch: bool = False,
 ) -> Iterator[Any]:
     # If all trigger channels subscribed by this process are not empty
     # then invoke the process with the values of all non-empty channels
     if isinstance(proc.channels, dict):
         try:
             val: dict = {
                 k: read_channel(
-                    channels, chan, catch=catch or chan not in proc.triggers
+                    channels,
+                    chan,
+                    catch=chan not in proc.triggers,
```

Comment:
1. Because you can say subscribe to channels a and b, but when they are triggered also read whatever c has at the time (but don't be triggered if only c changed)
2. Because a list of channels to read is interpreted as "I want to read the value of the first of these channels that isn't empty"

Reply:
What's an example of reading from c but only subscribed to a and b?

Reply:
ah right - so channels are both state values as well as edges